packages/drivers/odsp-driver/src/prefetchLatestSnapshot.ts

```diff
@@ -0,0 +1,207 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { v4 as uuid } from "uuid";
+import { assert } from "@fluidframework/common-utils";
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+import { IResolvedUrl } from "@fluidframework/driver-definitions";
+import {
+    ICacheEntry,
+    IOdspResolvedUrl,
+    IPersistedCache,
+    ISnapshotOptions,
+    OdspResourceTokenFetchOptions,
+    snapshotKey,
+    TokenFetcher,
+} from "@fluidframework/odsp-driver-definitions";
+import { ChildLogger, PerformanceEvent, TelemetryLogger } from "@fluidframework/telemetry-utils";
+import { LocalPersistentCache } from "./odspCache";
+import { ISnapshotCacheValue } from "./odspDocumentStorageManager";
+import {
+    createOdspLogger,
+    fetchAndParseAsJSONHelper,
+    getOdspResolvedUrl,
+    getWithRetryForTokenRefresh,
+    IOdspResponse,
+} from "./odspUtils";
+import { IOdspSnapshot } from "./contracts";
+import { evalBlobsAndTrees, toInstrumentedOdspTokenFetcher } from "./odspUtils2";
+
+/**
+ * Function to prefetch the snapshot and cached it in the persistant cache, so that when the container is loaded
+ * the cached latest snapshot could be used and removes the network call from the critical path.
+ * @param resolvedUrl - Resolved url to fetch the snapshot.
+ * @param getStorageToken - function that can provide the storage token for a given site. This is
+ *  is also referred to as the "VROOM" token in SPO.
+ * @param persistedCache - Cache to store the fetched snapshot.
+ * @param logger - Logger to have telemetry events.
+ * @param hostSnapshotFetchOptions - Options to fetch the snapshot if any. Otherwise default will be used.
+ * @returns - True if the snapshot is cached, false otherwise.
+ */
+export async function prefetchLatestSnapshot(
+    resolvedUrl: IResolvedUrl,
+    getStorageToken: TokenFetcher<OdspResourceTokenFetchOptions>,
+    persistedCache: IPersistedCache = new LocalPersistentCache(),
```

Comment:
What does setting a default `= new LocalPersistentCache()` mean in the context of this parameter being followed by a non-optional parameter?

And under what circumstances would hosts provide their own?

Reply:
It feels like a default implementation would also increase bundle size unnecessarily, in situations where the host overrides it anyway

Reply:
Yes, default is useless here as it will defeat the purpose of this API (as cache would be in memory and would destroy after the call). I missed that :)